spec/lib/api/2.0/profiles-spec.js

```diff
@@ -0,0 +1,373 @@
+        sinon.stub(taskProtocol, 'requestCommands').resolves({ testcommands: 'cmd' });
+        sinon.stub(taskProtocol, 'requestProfile').resolves();
+        sinon.stub(taskProtocol, 'requestProperties').resolves();
+
+        workflowApiService = helper.injector.get('Http.Services.Api.Workflows');
+        sinon.stub(workflowApiService, 'findActiveGraphForTarget').resolves({});
+        sinon.stub(workflowApiService, 'createActiveGraph').resolves({ instanceId: 'test' });
+
+        presenter = helper.injector.get('common-api-presenter');
+
+        profiles = helper.injector.get('Profiles');
+        sinon.stub(profiles, 'getAll').resolves();
+        sinon.stub(profiles, 'getName').resolves();
+        sinon.stub(profiles, 'get').resolves();
+        sinon.stub(profiles, 'put').resolves();
+
+        // add southbound routes here
+
+        profileApiService = helper.injector.get('Http.Services.Api.Profiles');
+        sinon.stub(profileApiService, 'getNode').resolves({});
+        sinon.stub(profileApiService, 'createNodeAndRunDiscovery').resolves({});
+        sinon.stub(profileApiService, 'runDiscovery').resolves({});
+        sinon.stub(profileApiService, 'setLookup').resolves();
+
+        waterline = helper.injector.get('Services.Waterline');
+        sinon.stub(waterline.lookups, "findOneByTerm").resolves();
+
+        return helper.injector.get('Views').load();
+    });
+
+    afterEach('teardown mocks', function () {
+        function resetMocks(obj) {
+            _(obj).methods().forEach(function (method) {
+                if (typeof obj[method].restore === 'function') {
+                    obj[method].restore();
+                }
+            }).value();
+        }
+        resetMocks(lookupService);
+        resetMocks(taskProtocol);
+        resetMocks(workflowApiService);
+        resetMocks(presenter.CommonApiPresenter.prototype);
+        resetMocks(profiles);
+        resetMocks(profileApiService);
+        resetMocks(waterline.lookups);
+    });
+
+    after('stop HTTP server', function () {
+        return helper.stopServer();
+    });
+
+    var profile = [{
+        id: '1234abcd5678effe9012dcba',
+        name: 'dummy_profile',
+        contents: '#!ipxe\n',
+        hash: '123',
+        scope: 'dummy'
+    }];
+
+    describe('2.0 GET /profiles/metadata', function () {
+        it('should return a list of profiles', function () {
+            profiles.getAll.resolves(profile);
+            return helper.request().get('/api/2.0/profiles/metadata')
+                .expect('Content-Type', /^application\/json/)
+                .expect(200)
+                .then(function(res) {
+                    expect(res.body[0]).to.have.property('name', 'dummy_profile');
+                    expect(res.body[0]).to.have.property('scope', 'dummy');
+		            expect(res.body[0]).to.have.property('hash', '123');
+		            expect(res.body[0]).to.not.have.property('blank');
```

Comment:
A little odd to me, why expect not have property `blank`? If you want to expect the response has no extra properties, you need to check all keys of res.body[0] to exactly equal to a list. such as:
```javascript
expect(_.keys(res.body[0])).to.deep.equal(['key1', 'key2', ...]);
```